lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -453,6 +453,18 @@ def _extract_collection_from_git(repo_url, coll_ver, b_path):
             proc_err,
         )
 
+    if pathlib.Path(str(b_checkout_path) + "/.gitmodules").is_file():
```

Comment:
Also, I don't think that doing this unconditionally would be a great idea. If there's gitmodules entries pointing at private or password-protected Git remotes, this would fail. So it'd probably be reasonable to make it depend on a CLI option:
```suggestion
    has_submodules = (pathlib.Path(b_checkout_path) / '.gitmodules').is_file()
    with_submodules_cli_request = ...
    if with_submodules_cli_request and has_submodules:
```

Reply:
```python
    has_submodules = (pathlib.Path(b_checkout_path.decode()) / '.gitmodules').is_file()
    with_submodule_cli_request = context.CLIARGS['init_submodules']
    if with_submodule_cli_request and has_submodules:
```

init_submodules is a boolean flag:

```
(.venv) ➜  ansible git:(devel) ✗ bin/ansible-galaxy collection install --help 2>/dev/null| grep init-submodules
                                         [--init-submodules]
  --init-submodules     Init collection submodules
```